basin3d/core/catalog.py

```diff
@@ -82,62 +108,61 @@ def initialize(self, plugin_list: list):
 
                 datasource = plugin.get_datasource()
                 self._process_plugin_attr_mapping(plugin, mapping_filename, datasource)
-                self._insert(datasource)
 
             logger.info(f"Initialized {self.__class__.__name__} metadata catalog ")
 
     # ---------------------------------
     # Methods to be over-witten by children classes
 
-    def _init_catalog(self):
+    def _init_catalog(self, **kwargs):
         """
+        Initialize the catalog database
 
-        :return:
         """
         raise NotImplementedError
 
     def _insert(self, record):
         """
         Insert the record
-        :param record:
+
+        :param record: the record to insert
         """
         raise NotImplementedError
 
-    def _get_datasource(self, datasource_id) -> Optional[DataSource]:
+    def _get_datasource(self, datasource_id) -> Optional[basin3d.core.models.DataSource]:
```

Comment:
Do we still need this method?

Reply:
We can consider this in the future but since it is part of the class definition which is used by the Django we need to make sure we don't break anything.